src/tests/ftest/interoperability/upgrade_downgrade_base.py

```diff
@@ -130,594 +185,581 @@ def show_daos_version(self, all_hosts, hosts_client):
+            for package in ['daos-client-tests']]
+
+        # Make sure the installed libfabric matches the shipped versions
+        # TODO need a more proper and generic solution
+        #      MAYBE "dnf --repo=daos-packages-2.2.0 list --available"
+        #      MAYBE python "dnf" module
+        # TODO this doesn't work because of libfabric incompatibility
+        # if Version(version) == '2.2.0':
+        #     packages.append('libfabric-1.15.1-3.el8')
+        # elif Version(version) == '2.4.1':
+        #     packages.append('libfabric-1.19.0-1.el8')
+
+        server_packages = packages.copy()
+        client_packages = packages + ['ior']
 
-        Args:
-            servers (NodeSet): servers to be upgraded.
-            clients (NodeSet): clients to be upgraded.
-        """
-        if ".repo" in self.upgrade_repo:
-            repo_1 = self.upgrade_repo
-            repo_2 = self.downgrade_repo
-            self.updowngrade_via_repo(servers, clients, repo_1, repo_2)
-        else:
-            all_hosts = servers + clients
-            self.updowngrade_via_rpms(all_hosts, "downgrade", self.downgrade_repo)
-
-    def updowngrade_via_rpms(self, hosts, updown, rpms):
-        """Upgrade downgrade hosts
+        if servers:
+            self.log.info("Installing version %s on servers, %s", version, servers)
+            if not install_packages(self.log, servers, server_packages, 'root').passed:
+                self.fail(f"Failed to install version {version} on servers")
+            self.current_server_version = Version(version)
+            result = run_remote(self.log, NodeSet(servers[0]), 'dmg version')
+            if not result.passed:
+                self.fail(f"dmg version failed after installing server version {version}")
+            if not result.homogeneous:
+                self.fail(f"dmg version inconsistent after installing server version {version}")
+            self.log.info("Successfully installed version %s on servers", version)
+
+        # Install on clients
+        if clients:
+            self.log.info("Installing version %s on clients, %s", version, clients)
+            if not install_packages(self.log, clients, client_packages, 'root').passed:
+                self.fail(f"Failed to install version {version} on clients")
+            self.current_client_version = Version(version)
+            result = run_remote(self.log, clients, 'daos version')
+            if not result.passed:
+                self.fail(f"daos version failed after installing client version {version}")
+            if not result.homogeneous:
+                self.fail(f"daos version inconsistent after installing client version {version}")
+            self.log.info("Successfully installed version %s on clients", version)
+
+        # Restart servers and/or agents
+        if servers:
+            self.log.info("Restarting servers after installing %s", version)
+            errors = self.restart_servers(stop=False)
```

Comment:
Maybe we can use DaosServerManager.restart() here instead

Reply:
DaosServerManager.restart() works here. The downside is that it does not return errors or have an upper bound timeout. But that can be improved later

Reply:
Actually I messed something up testing this and need to try again